src/main/java/io/appium/java_client/MobileBy.java

```diff
@@ -51,7 +51,15 @@ public static By AccessibilityId(final String id) {
 
         return new ByAccessibilityId(id);
     }
-
+    
+    public static By IosNsPredicateString(final String iOSNsPredicateString) {
+    	if (iOSNsPredicateString == null) {
+    	  throw new IllegalArgumentException("Must supply an iOS NsPredicate String");
```

Comment:
Since this is just a comment I would format it "NSPredicate" instead of "NsPredicate"